esp/services/ws_workunits/ws_workunitsQuerySets.cpp

```diff
@@ -3299,3 +3299,100 @@ bool CWsWorkunitsEx::onWUGetNumFileToCopy(IEspContext& context, IEspWUGetNumFile
     }
     return true;
 }
+
+void getSummaryStatsByQueryId(const char *target, const char *queryId, const char *fromTime, const char *toTime, IArrayOf<IEspQuerySummaryStats>& querySummaryStatsList)
+{
+    if (!target || !*target)
+        throw MakeStringException(ECLWATCH_MISSING_PARAMS, "Target name required");
+    if (!queryId || !*queryId)
+        throw MakeStringException(ECLWATCH_MISSING_PARAMS, "Query Id required");
+
+    Owned<IConstWUClusterInfo> info = getTargetClusterInfo(target);
+    if (!info || (info->getPlatform()!=RoxieCluster)) //Only support roxie for now
+        throw MakeStringException(ECLWATCH_INVALID_CLUSTER_NAME, "Invalid Roxie name");
+
+    PROGLOG("getSummaryStatsByQueryId: target %s, query %s", target, queryId);
+
+    const SocketEndpointArray &eps = info->getRoxieServers();
+    if (eps.empty())
+        return;
+
+    VStringBuffer control("<control:queryAggregates");
+    if (!isEmpty(fromTime))
+        control.appendf(" from='%s'", fromTime);
+    if (!isEmpty(toTime))
+        control.appendf(" to='%s'", toTime);
+    control.appendf("><Query id='%s'/></control:queryAggregates>", queryId);
+    Owned<IPropertyTree> queryAggregates = sendRoxieControlAllNodes(eps.item(0), control.str(), false, ROXIELOCKCONNECTIONTIMEOUT);
+    if (!queryAggregates)
+        return;
+
+    if (getEspLogLevel() >= LogNormal)
+    {
+        StringBuffer sb;
+        toXML(queryAggregates, sb);
+        DBGLOG("getSummaryStatsByQueryId(): <%s> <%s>", control.str(), sb.str());
```

Comment:
is this big?  Will it help debug something if we log it even at logmax?  if so, maybe if (getEspLogLevel() >= LogMax)?

Also interesting format... putting the xml blob inside <>?